### PR TITLE
Fix emission of enter and leave events

### LINKED ISSUE #1259: [BUG] Page, bezel swipe back triggers enter event
**Describe the bug**

Bezel swipe back triggers enter event on KirbyPage. Enter is not triggered on back button click.

**To Reproduce**

1. Go to eg. transaction details page
2. Make bezel swipe back to the transaction list
3. See that enter event is triggered

**Kirby version**
- [e.g. 1.1.10]

**Please complete the following information:**
- Chrome iOS emulator

**Additional context**

## Tasks:
- [ ] Move issue from "Ready to do" -> "in progress" on our [project board](https://github.com/kirbydesign/designsystem/projects/1)
- [ ] Create Fix Branch
- [ ] Create test reproducing the error
- [ ] Fix
- [ ] If relevant [Update Examples and Showcase](https://cookbook.kirby.design/home/showcase/button)
- [ ] Push commits to GitHub
- [ ] Open a pull request (remember to add `closes #issueno` to the description) and ask @kirbydesign/kirby-guild for review
- [ ] Merge to master and delete branch

### LINKED ISSUE #2259: [BUG] List Swipe is missing pressed state for `ion-options`
Not sure this is even a thing...

image

...and what about hover?

### PR DESCRIPTION
**Pull request by mictro**
## Which issue does this PR close?

This PR closes #1259 

## What is the new behavior?

Fix wrong emissions of page enter and leave on kirby-page when navigating between multiple components. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [N/A] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ N/A] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/src/lib/components/page/page.component.spec.ts [hunk 1]
```diff
@@ -52,11 +54,15 @@ describe('PageComponent', () => {
...
+          path: pageUrl,
           component: PageComponent,
         },
         {
-          path: 'someUrl',
+          path: firstOtherUrl,
+          component: DummyComponent,
+        },
+        {
+          path: secondOtherUrl,
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/page/page.component.spec.ts:65]: Could these be Page components instead, so we make sure to test the stacked behavior? We would have to refactor the enter/leave tests below a tiny bit to account for additional enter/leaves being called when navigating.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Awesome work, looks great. A couple of test-related comments is all I got.

File: libs/designsystem/src/lib/components/page/page.component.spec.ts (hunk 1 above)
Comment (mictro) [reply libs/designsystem/src/lib/components/page/page.component.spec.ts]: Good point - done

File: libs/designsystem/src/lib/components/page/page.component.ts [hunk 2]
```diff
@@ -273,18 +273,23 @@ export class PageComponent
   }
 
   ngAfterViewInit(): void {
+    // This instance has observed a page enter
```
[the remark this answers is not in the document]
Comment (mictro) [reply libs/designsystem/src/lib/components/page/page.component.ts]: Yes indeed - done

File: libs/designsystem/src/lib/components/page/page.component.ts [hunk 3]
```diff
@@ -335,11 +335,13 @@ export class PageComponent
   }
 
   private onLeave() {
+    if (!this.hasEntered) return;
```
[the remark this answers is not in the document]
Comment (mictro) [reply libs/designsystem/src/lib/components/page/page.component.ts]: done!

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**